libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Keychain/KeychainManager.swift

```diff
@@ -0,0 +1,272 @@
+//  * Redistributions of source code must retain the above copyright notice, this list of conditions
+//  and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//  conditions and the following disclaimer in the documentation and/or other materials provided
+//  with the distribution.
+//  * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+//  endorse or promote products derived from this software without specific prior written
+//  permission of salesforce.com, inc.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+//  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+//  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import Foundation
+
+internal protocol KeychainItemQueryable {
+    var query: [String: Any] { get }
+}
+
+/// Adds kSecClassGenericPassword to the queryable
+internal struct GenericPasswordItemQuery: KeychainItemQueryable {
+    
+    let service: String
+    let account: String?
+    let accessGroup: String?
+    var query: [String: Any] {
+        var query: [String: Any] = [String(kSecClass): String(kSecClassGenericPassword),
+                                    String(kSecAttrService): service,
+                                    String(kSecAttrAccessible): kSecAttrAccessibleWhenUnlockedThisDeviceOnly]
+        
+        #if !targetEnvironment(simulator)
+        if let accessGroup = accessGroup {
+            query[String(kSecAttrAccessGroup)] = accessGroup
+        }
+        #endif
+        if let account = account {
+            query[String(kSecAttrAccount)] = account
+        }
+        
+        return query
+    }
+    
+}
+
+internal class KeychainItemManager: NSObject {
+    private let secureStoreQueryable: KeychainItemQueryable
+    static let errorDomain = "com.salesforce.security.keychainException"
+    
+    /// Initializer for kSecClassGenericPassword with accessgroup. Will create a keychain item manager for kSecClassGenericPassword operations
+    init(service: String, account: String?) {
+        self.secureStoreQueryable = GenericPasswordItemQuery(service: service,
+                                                             account: account,
+                                                             accessGroup: nil)
```

Comment:
Should this one simply call the initializer below?

Reply:
actually we don't need the other initializer for now. I will remove it.